### PR TITLE
DAOS-8900 obj: fix a typo in dc_obj_punch

### PR DESCRIPTION
**Pull request by liuxuezhao**
Fix a typo of punch opcode in dc_obj_punch, and a few other refinings.

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: src/object/cli_obj.c
Comment (Nasf-Fan) [inline src/object/cli_obj.c:5205]: Thanks for catching that.

File: src/tests/suite/daos_epoch_io.c
Comment (Nasf-Fan) [inline src/tests/suite/daos_epoch_io.c:345]: Why not move the rebuild_wait() after reint_server()? Do we need to wait rebuild for reintegration case?

Comment (liuxuezhao) [reply src/tests/suite/daos_epoch_io.c:345]: probably fine to keep the rebuild wait here.
I just want to add the rebuild wait after exclude, because in daos_run_io_conf test there are lots of exlude+reint, that may cause co-located layout, so if we exclude one rank here that may cause multi-shards failure, that may cause degraded fetch failure.

Comment (liuxuezhao) [reply src/tests/suite/daos_epoch_io.c:345]: i updated it a little bit.

**Review by Nasf-Fan — APPROVED** [review]

**Comment by kprantis** [comment]
Looks like this one is passed test and just waiting on another review?

**Review by wangdi1 — APPROVED** [review]

**Comment by kprantis** [comment]
Ready for gatekeeper?

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**
